client/modules/IDE/actions/projects.js

```diff
@@ -8,6 +8,7 @@ const ROOT_URL = __process.env.API_URL;
 
 export function getProjects(username) {
   return (dispatch) => {
+    dispatch({ type: ActionTypes.START_LOADING });
```

Comment:
rather than putting the action payload inline, this should call a function. same with the other times dispatch is called.

Reply:
Something like startLoading and stopLoading ? Also should I put these functions in a separate file?

Reply:
yes to both questions! every reducer should have its own actions/reducer file.